File: Lecture04_Navigation/AfterLecture_Navigation_Java/app/src/main/java/no/hiof/larseknu/playingwithnavigation/HomeFragment.java
package no.hiof.larseknu.playingwithnavigation;


import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;


/**
 * A simple {@link Fragment} subclass.
 */
public class HomeFragment extends Fragment {


    public HomeFragment() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_home, container, false);
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        // Get the bundle with the arguments
        Bundle arguments = getArguments();

        // Checks if we actuelly get some arguments
        if (arguments != null) {
            // Retrieves a reference to the TextView
            TextView usernameTextView = view.findViewById(R.id.textViewUsername);

            // Retrieves the username from arguments with the help of the Safe Args generated file HomeFragmentArgs
            String username = HomeFragmentArgs.fromBundle(arguments).getUsername();

            // Sets the username
            usernameTextView.setText(username);
        }
    }
}

File: Lecture04_Navigation/AfterLecture_Navigation_Kotlin/app/src/main/java/no/hiof/larseknu/playingwithnavigation/HomeFragment.kt
package no.hiof.larseknu.playingwithnavigation


import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.navigation.fragment.navArgs

/**
 * A simple [Fragment] subclass.
 */
class HomeFragment : Fragment() {

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_home, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        // Checks if we actually get some arguments
        if (arguments != null) {
            // Retreives a reference to the TextView
            val usernameTextView: TextView = view.findViewById(R.id.textViewUsername)

            // Retreives the username from arguments through the Safe Args generated file HomeFragmentArgs
            val username = HomeFragmentArgs.fromBundle(arguments!!).username

            // Sets the username
            usernameTextView.text = username
        }
    }
}
